airflow/utils/session.py

```diff
@@ -29,7 +29,9 @@ def create_session():
     session = settings.Session()
     try:
         yield session
-        session.commit()
+        # Only Commit is a new or a modified object exists in the session
+        if not session._is_clean():  # pylint: disable=protected-access
```

Comment:
Both the file changes are not strictly needed. Either of the two file changes are sufficient.

In one cases we commit an empty session and in other case we don't commit a session if it is empty.